Assets/_Scripts/3Managers/BaseBuildingValidator.cs

```diff
@@ -1,29 +1,78 @@
 ﻿using OrcaDefense.Models;
 using System;
+using System.Collections.Generic;
 
 public class BaseBuildingValidator : IBuildingValidator
 {
-    public BaseBuildingValidator()
+    protected readonly Tile _tile;
+    public BaseBuildingValidator(Tile tile)
     {
+        _tile = tile;
     }
 
-    public virtual bool ShouldRenderRoof(Island map, int x, int y)
+    //Should this building render a roof?
+    public virtual bool ShouldRenderRoof(Island map)
     {
-        return !map.GetTileAt(x, y + 1).IsOccupied;
+        return map.Up(_tile.X, _tile.Y).Building is null;
     }
 
-    public virtual string[] ValidateAdjacencies(Island map, int x, int y)
+    //Does this building care if we build it next to it's neighbours?
+    public virtual string[] ValidateAdjacencies(Island map)
```

Comment:
List<string> instead? For all methods, currently it's a mix of arrays and lists.

Reply:
Agreed